app/src/main/java/com/arygm/quickfix/ui/search/Announcement.kt

```diff
@@ -1,33 +1,308 @@
+
+            QuickFixButtonWithIcon(
+                buttonText = "Upload pictures",
+                onClickAction = {
+                  // TODO: Apply the backend of the pictures
+                },
+                buttonColor = colorScheme.surface,
+                textColor = colorScheme.onBackground,
+                textStyle = MaterialTheme.typography.titleMedium,
+                modifier =
+                    Modifier.width(360.dp)
+                        .height(90.dp)
+                        .testTag("picturesButton")
+                        .graphicsLayer(alpha = 1f),
+                iconId = R.drawable.upload_image,
+                iconContentDescription = "upload_image",
+                iconColor = colorScheme.onBackground)
+            Row(
+                modifier = Modifier.fillMaxWidth().padding(vertical = 8.dp).padding(start = 8.dp),
+            ) {
               Text(
-                  text = "Welcome to the Announcement Screen",
-                  modifier = Modifier.padding(padding).testTag("AnnouncementText"))
+                  text = "* Mandatory fields",
+                  color =
+                      if (titleIsEmpty ||
+                          !categoryIsSelected ||
+                          !locationIsSelected ||
+                          descriptionIsEmpty)
+                          colorScheme.error
+                      else colorScheme.onSecondaryContainer,
+                  style = MaterialTheme.typography.bodySmall,
+                  modifier = Modifier.padding(start = 9.dp).testTag("mandatoryText"))
+            }
+
+            QuickFixButton(
+                buttonText = "Post your announcement",
+                onClickAction = {
+                  // TODO: Apply the backend of the announcement creation
+                },
+                buttonColor = colorScheme.primary,
+                textColor = colorScheme.onPrimary,
+                textStyle = MaterialTheme.typography.titleMedium,
+                modifier =
+                    Modifier.width(360.dp)
+                        .height(55.dp)
+                        .testTag("announcementButton")
+                        .graphicsLayer(alpha = 1f),
+                enabled =
+                    !titleIsEmpty &&
+                        categoryIsSelected &&
+                        locationIsSelected &&
+                        !descriptionIsEmpty)
+          }
+        })
+  }
+}
+
+@Composable
+fun QuickFixButtonWithIcon(
+    buttonText: String,
+    onClickAction: () -> Unit,
+    buttonColor: Color,
+    buttonOpacity: Float = 1f,
+    textColor: Color,
+    textStyle: TextStyle = MaterialTheme.typography.labelMedium,
+    horizontalArrangement: Arrangement.Horizontal = Arrangement.Center,
+    contentPadding: PaddingValues = ButtonDefaults.ContentPadding,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    iconId: Int,
+    iconContentDescription: String,
+    iconColor: Color
+) {
+  Button(
```

Comment:
Again here Why not using QuickFixButton instead of a button, it's just code duplication.
Anyways you can look at #129 and take the new QuickFixButton, it contains the new updates with leading and trailing icons